amaranth/hdl/_ast.py

```diff
@@ -2052,7 +2052,9 @@ def __init__(self, shape=None, *, name=None, init=None, reset=None, reset_less=F
                             .format(orig_init, shape),
                     category=SyntaxWarning,
                     stacklevel=2)
-        self._init = init.value
+        self._init = init.value & ((1 << self._width) - 1)
```

Comment:
it may be a better idea to instead use `Const(..., shape).value`, and do it near the `init = Const.cast(init)` branch above?

Reply:
I'm not sure what you mean by the second part of this sentence. Can you write it as a diff?